src/lib/client.ts

```diff
@@ -65,8 +65,19 @@ export class HTTPClient implements IRequestClient {
 
         axiosInstance.interceptors.response.use(
             (response) => response,
-            // eslint-disable-next-line @typescript-eslint/promise-function-async
-            (error) => Promise.reject(error.response.data),
+            // eslint-disable-next-line @typescript-eslint/promise-function-async,@typescript-eslint/tslint/config
+            (error) => {
+                if (!!error.response) {
+                    return Promise.reject(error.response.data);
+                } else if (!!error.request) {
+                    return Promise.reject(error.request);
```

Comment:
Did a small search and stumbled on:
> A pending promise can either be fulfilled with a value or rejected with a reason (error).

(source: [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#description))

and:

https://eslint.org/docs/rules/prefer-promise-reject-errors which suggests to use `Error` objects for user-defined errors in Promises. I do think some of the reasoning supplied there also holds for errors that aren't user-defined.

Reply:
At the beginning I was thinking that maybe it would be useful to provide something of the request, since the entire object may be too vague and big for a user to understand what went wrong. But I couldn't find a way to make it better than the object itself.

Now with your mention, I start to believe that the request itself does not entitle as a `reason` for rejecting a promise. Therefore, it should contain the message, along with the request to make it somehow better.

Reply:
Let's keep this simple:

The goal is to pull the HTTP response body out of the `error.response` object, if it exists.

If the error doesn't come from the server, let's return the entire error object: `Promise.reject(error)`. Simple as that :) 

We should try to use built-in error handling as much as possible.